pkg/controllers/mcs/service_export_controller.go

```diff
@@ -424,12 +424,15 @@ func (c *ServiceExportController) removeOrphanWork(endpointSliceObjects []runtim
 
 	collectedEpsWorkList := &workv1alpha1.WorkList{}
 	if err := c.List(context.TODO(), collectedEpsWorkList, &client.ListOptions{
-		Namespace: names.GenerateExecutionSpaceName(targetCluster),
+		Namespace: names.GenerateExecutionSpaceName(serviceExportKey.Cluster),
 		LabelSelector: labels.SelectorFromSet(labels.Set{
 			util.PropagationInstruction: util.PropagationInstructionSuppressed,
+			util.ServiceNamespaceLabel:  serviceExportKey.Namespace,
+			util.ServiceNameLabel:       serviceExportKey.Name,
```

Comment:
This is the key point that fetched Works do not belong to the current service, and these Works will be accidentally deleted.